### DISCUSSION TITLE
How to access the translated version of a page that has a translated slug

### DISCUSSION THREAD

**Discussion by carlosnasillo** [wagtail/wagtail-localize#308] [Q&A] [answered]
Typically with common paths on Django the following works fine: 

```
translation.activate('en')
urls.translate_url("https://example.com/blog/", "en")
>> https://example.com/blog/
urls.translate_url("blog", "es")
>> https://example.com/es/blog/
```
But what if we translated the slug? for example of a blog post `/page-1/` to be in `es` = `/pagina-1/`

**Desired output**
```
>> https://example.com/blog/page-1
>> https://example.com/es/blog/pagina-1
```

If I try the above again with a translated slug URL, I get `https://example.com/blog/page-1` obviously Django is not able to resolve it so defaults to base language. Is there a different way in which I should be doing it? 

Thanks again for all the help, new to wagtail-localize, wagtail and i18n - so all a bit overwhelming ATM.

**Comment by kaedroho** [comment, marked as answer]
Hey @carlosnasillo 

You can't use Django's ``translate_url`` for this because it's unaware of Wagtail's own routing.

Instead, you need to get the translated page instance and access its ``.url`` attribute. Wagtail will make sure the URL has the correct language prefix.

For example:

```python
from wagtail.core.models import Locale


spanish = Locale.objects.get(language_code='es')
spanish_blog_page = english_blog_page.get_translation(spanish)

if spanish_blog_page.live:
    return spanish_blog_page.url
```

**Reply by carlosnasillo** [reply]
Thank you very much the guidance @kaedroho, super appreciated!
